packages/@aws-cdk/aws-appmesh/lib/gateway-route.ts

```diff
@@ -26,6 +27,72 @@ export interface IGatewayRoute extends cdk.IResource {
    * The VirtualGateway the GatewayRoute belongs to
    */
   readonly virtualGateway: IVirtualGateway;
+
+  /**
+   * Grants the given entity all read permissions for this GatewayRoute.
+   */
+  grantRead(identity: iam.IGrantable): iam.Grant;
+
+  /**
+   * Grants the given entity all write permissions for this GatewayRoute.
+   */
+  grantWrite(identity: iam.IGrantable): iam.Grant;
+}
+
+abstract class GatewayRouteBase extends cdk.Resource implements IGatewayRoute {
+  /**
+   * The name of the GatewayRoute
+   *
+   * @attribute
+   */
+  public abstract readonly gatewayRouteName: string;
+
+  /**
+   * The Amazon Resource Name (ARN) for the GatewayRoute
+   *
+   * @attribute
+   */
+  public abstract readonly gatewayRouteArn: string;
+
+  /**
+   * The VirtualGateway the GatewayRoute belongs to
+   */
+  public abstract readonly virtualGateway: IVirtualGateway;
+
+  /**
+   * Grants the given entity all read permissions for this VirtualGateway.
+   */
+  public grantRead(identity: iam.IGrantable): iam.Grant {
+    return this.grant(identity,
+      'appmesh:DescribeGatewayRoute',
+      'appmesh:ListGatewayRoute',
+      'appmesh:ListTagsForResource',
+    );
+  }
+
+  /**
+   * Grants the given entity all write permissions for this VirtualGateway.
+   */
+  public grantWrite(identity: iam.IGrantable): iam.Grant {
+    return this.grant(identity,
+      'appmesh:CreateGatewayRoute',
+      'appmesh:UpdateGatewayRoute',
+      'appmesh:DeleteGatewayRoute',
+      'appmesh:TagResource',
+      'appmesh:UntagResource',
```

Comment:
I have to say, I'm a little surprised by these permissions here.

This is the `GatewayRoute` resource itself. Why would be allow creating and deleting it to other resources? I understand that the IAM principal performing the CloudFormation deployment needs these permissions... but why would a different principal from the same Stack as this Gateway Route be allowed to delete it?

@dfezzie  What's the use case for having these permissions?

Reply:
I'm a little confused on what you mean. Can you explain what the purpose of IAM grants is in general? I thought we needed to provide these base permission groups for our resources to be considered feature complete for CDK. How do other teams typically implement IAM grants?

Reply:
Yeah, I think there's some confusion here 🙂. Let's look at the existing `grant*()` APIs as a reference, maybe. that will clears things up.

Let's take S3 as an example. when you call `bucket.grantWrite(principal)`, it grants the permissions `s3:PutObject` and `s3:Abort*` on that specific Bucket. These are used by resources that write to the Bucket.

But note that it _doesn't_ grant `s3:CreateBucket`, nor `s3:DeleteBucket`!

1. The Bucket obviously already exists, so there's no point in granting `s3:CreateBucket` on its ARN.
2. The resource should write things to the Bucket -it should _not_ have permissions to delete the entire Bucket! Only the deployment Role needs those permissions.

> I thought we needed to provide these base permission groups for our resources to be considered feature complete for CDK. How do other teams typically implement IAM grants?

No. There is no such requirement. There are many, many resources in the CDK that do _not_ have `grant*()` methods, because they don't make sense for that resources (for example, what would `grantRead()` mean for a Load Balancer?).

----------------

Does this make sense @dfezzie?

Reply:
Yeah, that does make sense. 

So, with that in mind, I think the _most_ applicable IAM Grant for us is `grantStreamAggregatedResources`. This is needed for granting whatever compute your mesh services are running on to fetch the config from our Envoy Management Service. 

I could also imagine a scenario where you would need to grant high level admin permissions to manage cluster resources to some role. For example, we have an `App Mesh Controller` that manages the lifecycle of App Mesh objects in K8s. The master node there would need those permissions, but that doesn't work with the current approach in this PR.

@alexbrjo what are your thoughts on this?

Reply:
I included create and delete simply because they are "write" actions.

An update (which can break routing configuration) is on the same level of access as a delete or a create (which in this case would be a re-creation of a previously existing resource). The only benefit of excluding them is if the customer use case is to grant the ability to modify a resource but safe guard against a deletion, but this is speculation.

Reply:
Thanks for explaining this, Adam. I thought we wanted to include all actions for each resource. In this case I don't think it makes sense to include control plane actions unless we're doing a recursive `grantAdmin` control of the mesh for k8s controller. I think we could go with only `grantStreamAggregatedResources`.

Reply:
Sounds good @alexbrjo - let's go with only `grantStreamAggregatedResources()`.